lib/prefix.c

```diff
@@ -1468,11 +1468,11 @@ in_addr_t ipv4_broadcast_addr(in_addr_t hostaddr, int masklen)
 
 	masklen2ip(masklen, &mask);
 	return (masklen != IPV4_MAX_PREFIXLEN - 1) ?
-						   /* normal case */
-		       (hostaddr | ~mask.s_addr)
-						   :
-						   /* special case for /31 */
-		       (hostaddr ^ ~mask.s_addr);
+		/* normal case */
+		(hostaddr | ~mask.s_addr)
+		   :
+		/* For prefix 31 return 255.255.255.255 (RFC3021) */
+		htonl(0xFFFFFFFF);;
```

Comment:
double semicolon?

Reply:
A typo. Thanks for spotting this. Fixed.